packages/app/src/cli/services/app/write-app-configuration-file.ts

```diff
@@ -33,6 +33,9 @@ export const rewriteConfiguration = <T extends zod.ZodTypeAny>(schema: T, config
   if (schema instanceof zod.ZodArray) {
     return (config as unknown[]).map((item) => rewriteConfiguration(schema.element, item))
   }
+  if (schema instanceof zod.ZodEffects) {
+    return rewriteConfiguration(schema._def.schema, config)
+  }
```

Comment:
@gonzaloriestra what is this doing, thanks for fixing these tests!

Reply:
This function is iterating through the elements of the config and checking the types to do different things depending on that. And it looks like the new WebhooksSchemaWithDeclarative is not a ZodObject anymore, but a [ZodEffects](https://github.com/colinhacks/zod#zodtype-with-zodeffects) because of the refining, which is a wrapper with more stuff. So I had to take that into account to keep iterating through the inner objects there.